libs/net/dal/Migrations/1.0.31/Up/PostUp/00-Report.sql

```diff
@@ -0,0 +1,148 @@
+DO $$
+BEGIN
+
+UPDATE public.report SET
+  "filter" = '
+    {
+      "size": 500,
+      "query": {
+        "bool": {
+          "must": [
+            {
+              "range": {
+                "publishedOn": {
+                  "gte": "now-1d/d"
+                }
+              }
+            },
+            {
+              "terms": {
+                "status": [
+                  "Publish",
+                  "Published"
+                ]
+              }
+            },
+            {
+              "terms": {
+                "contentType": [
+                  "PrintContent",
+                  "Image"
+                ]
+              }
+            },
+            {
+              "terms": {
+                "isHidden": [
+                  false
+                ]
+              }
+            },
+            {
+              "bool": {
+                "should": [
+                  {
+                    "terms": {
+                      "sourceId": [
+                        91,
+                        246,
+                        247,
+                        248,
+                        249
+                      ]
+                    }
+                  }
+                ]
+              }
+            }
+          ]
+        }
+      }
+    }'::jsonb,
+    "template" = '@inherits RazorEngineCore.RazorEngineTemplateBase<TNO.Services.Reporting.Models.TemplateModel>
+@using System
+@using System.Linq
+@using TNO.Entities
+@{
+    var groups = Content.GroupBy(
+      c => $"{c.Source?.Name}",
+      c => c,
+      (k, c) => new { Key = k, Content = c }).OrderBy(c => c.Key);
+
+    var content = Content.ToArray();
+    var frontPages = content.Where(x => x.ContentType == ContentType.Image);
+    var today = System.DateTime.Now.ToString("dd-MMMM-yyyy");
+}
+<h2 id="top">Today''s News Online - Morning Report</h2>
+<div style="color:red;">DO NOT FORWARD THIS EMAIL TO ANYONE</div>
+<br />
+<div>@System.DateTime.Now.ToString("dddd, MMMM d, yyyy")</div>
+<br />
+<h3>Table of Contents</h3>
+<ul>
+    <li><a href="#front-pages">Front pages</a></li>
+</ul>
+@foreach (var group in groups)
+{
+    <ul>
+        <li><b>@group.Key</b></li>
+        @foreach (var c in group.Content)
+        {
+            var byline = !System.String.IsNullOrWhiteSpace(c.Byline) ? $" - {c.Byline}" : "";
+            var key = "item-" + c.Id;
+            <li><a href="#@key">@c.Headline @byline - @group.Key - @today</a></li>
+        }
+    </ul>
+}
+<hr />
+<a id="front-pages" href="#top">top</a>
+<h3 style="margin-top: 0.5rem;">Front Pages</h3>
+@foreach (var frontPage in frontPages)
+{
+    if (!string.IsNullOrEmpty(frontPage.ImageContent))
+    {
+        var src = $"data:{frontPage.ContentType};base64," + frontPage.ImageContent;
+        <img src="@src" alt="@frontPage.FileReferences.FirstOrDefault()?.FileName" />
+    }
+}
+@for (var i = 0; i < content.Length; i++)
```

Comment:
Won't this duplicate front page images in the loop?

Reply:
In this case it won't duplicate front page images unless there are more than one front page has the same image if that is what you meant.